src/particles/wakefields/WakeConvolution.cpp

```diff
@@ -0,0 +1,167 @@
+/* Copyright 2022-2023 The Regents of the University of California, through Lawrence
+ *           Berkeley National Laboratory (subject to receipt of any required
+ *           approvals from the U.S. Dept. of Energy). All rights reserved.
+ *
+ * This file is part of ImpactX.
+ *
+ * Authors: Alex Bojanich, Chad Mitchell, Axel Huebl
+ * License: BSD-3-Clause-LBNL
+ */
+#include "WakeConvolution.H"
+#include "particles/ImpactXParticleContainer.H" //Includes all necessary AMReX headers
+#include "initialization/InitDistribution.H"
+
+#ifdef ImpactX_USE_FFT
+#include <ablastr/math/fft/AnyFFT.H>
+#endif
+
+#include <algorithm>
+#include <cmath>
+#include <iostream>
+#include <stdexcept>
+#include <vector>
+
+/*
+Wake Functions:
+
+Define the longitudinal and transverse resistive wall wake functions,
+the longitudinal CSR wake function, and their associated functions
+
+*/
+
+//Step Function
+double unit_step(double s)
```

Comment:
As discussed: replacing here and in the other new files all `double` with `amrex::Real` allows us to compile for single precision (non-default).